src/backend/gl/gl_common.c

```diff
@@ -907,14 +916,25 @@ bool gl_init(struct gl_data *gd, session_t *ps) {
 	glUniformMatrix4fv(pml, 1, false, projection_matrix[0]);
 	glUseProgram(0);
 
-	// Set up the size of the back texture
-	gl_resize(gd, ps->root_width, ps->root_height);
-
+	// Set up the size and format of the back texture
 	glBindFramebuffer(GL_DRAW_FRAMEBUFFER, gd->back_fbo);
-	glFramebufferTexture2D(GL_DRAW_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D,
-	                       gd->back_texture, 0);
 	glDrawBuffer(GL_COLOR_ATTACHMENT0);
-	if (!gl_check_fb_complete(GL_FRAMEBUFFER)) {
+	const GLint *format = (const GLint[]){GL_RGB8, GL_RGBA8};
+	if (gd->dithered_present) {
+		format = (const GLint[]){GL_RGB16, GL_RGBA16};
+	}
```

Comment:
ASAN does not like how format is initialized and then used below for `gd->back_format`. Fails with *stack-use-after-scope* in line 927. (only when run with `--dithered-present`)
This is most likely a false-positive as the referenced value is an integer (`typedef int GLint;`) and used by value not by reference...

```suggestion
	const GLint *format = gd->dithered_present ? (const GLint[]){GL_RGB16, GL_RGBA16}
	                                           : (const GLint[]){GL_RGB8, GL_RGBA8};
```

Reply:
ah good catch!

Reply:
yeah technically this is illegal as the temporary went out of scope. in practice `(const GLint[])` is probably stored in `.rodata`. but i should fix this.